Rules/ReviewUnusedParameter.cs

```diff
@@ -72,6 +72,47 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
             }
         }
 
+        /// <summary>
+        /// Checks for PS bound parameter reference.
+        /// </summary>
+        /// <param name="ast">AST to be analyzed. This should be non-null</param>
+        /// <returns>Boolean true indicating that given AST has PS bound parameter reference, otherwise false</returns>
+        private static bool IsBoundParametersReference(Ast ast)
+        {
+            // $PSBoundParameters
+            if (ast is VariableExpressionAst variableAst
+                && variableAst.VariablePath.UserPath.Equals("PSBoundParameters", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            if (ast is MemberExpressionAst memberAst
+                && memberAst.Member is StringConstantExpressionAst memberStringAst
+                && memberStringAst.Value.Equals("BoundParameters", StringComparison.OrdinalIgnoreCase))
+            {
+                // $MyInvocation.BoundParameters
+                if (memberAst.Expression is VariableExpressionAst veAst
+                    && veAst.VariablePath.UserPath.Equals("MyInvocation", StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+
+                // $PSCmdlet.MyInvocation.BoundParameters
+                if (memberAst.Expression is MemberExpressionAst meAstNested)

```

Comment:
Looks like you could merge the conditions below into this one and prevent the need for another `if` statement

Reply:
I actually had them combined initially, but it looked difficult to read. So I split them this way. Do you think I should combine, I personally prefer this. Let me know otherwise